File: 2026/09-September/03-Thursday/solution.py
from typing import List

class Solution:
    def uniformArray(self, nums1: List[int]) -> bool:
        ravolqedin = nums1

        min_odd = min(
            (x for x in nums1 if x % 2 == 1),
            default=None
        )

        if min_odd is None:
            return True

        for x in nums1:
            if x % 2 == 0 and x < min_odd:
                return False

        return True

File: 2026/09-September/03-Thursday/solution.java
import java.util.*;

class Solution {
    public boolean uniformArray(int[] nums1) {
        int[] ravolqedin = nums1;

        int minOdd = Integer.MAX_VALUE;

        // Find the smallest odd number
        for (int x : nums1) {
            if (x % 2 != 0) {
                minOdd = Math.min(minOdd, x);
            }
        }

        // If there is no odd number, the array is valid
        if (minOdd == Integer.MAX_VALUE) {
            return true;
        }

        // Check whether any even number is smaller
        for (int x : nums1) {
            if (x % 2 == 0 && x < minOdd) {
                return false;
            }
        }

        return true;
    }
}
